### PR TITLE
Split the routing results to its own function.

### PR DESCRIPTION
**Pull request by dlundgren**
This allows the core logic of the RoutingMiddleware to be extended, without having to copy & paste the core logic.

### CODE REVIEW

File: Slim/Middleware/RoutingMiddleware.php [hunk 1]
```diff
@@ -101,4 +98,18 @@ public function performRouting(ServerRequestInterface $request): ServerRequestIn
...
         }
     }
+
+    /**
+     * Resolves the route from the given request
+     *
+     * @param  ServerRequestInterface $request
+     * @return RoutingResults
+     */
+    protected function resolveRouteFromRequest(ServerRequestInterface $request)
```
Comment (l0gicgate) [inline Slim/Middleware/RoutingMiddleware.php]: Let's add the return type `RoutingResults` and rename the method to `resolveRoutingResultsFromRequest()`

**Review by l0gicgate — CHANGES_REQUESTED** [review]
@adriansuter I don't foresee any breaking changes with this. Would you mind having a look when you have a minute?

File: Slim/Middleware/RoutingMiddleware.php (hunk 1 above)
Comment (dlundgren) [reply Slim/Middleware/RoutingMiddleware.php]: The method has been renamed, and return type added.

**Review by adriansuter — APPROVED** [review]

**Review by l0gicgate — APPROVED** [review]

**Comment by l0gicgate** [comment]
Thank you for your contribution @dlundgren!

**Pull request merged**